sunpy/io/tests/test_ana.py

```diff
@@ -75,3 +75,11 @@ def test_f32c():
     afilename = tempfile.NamedTemporaryFile().name
     with pytest.raises(RuntimeError):
         ana.write(afilename, img_f32, 'testcase', 1)
+
+
+@skip_ana
+def test_memmap():
+    afilename = tempfile.NamedTemporaryFile().name
+    ana.write(afilename, img_f32, 'testcase', 0)
+    img_f32c_rec = ana.read(afilename, memmap=True)
```

Comment:
Does the ana reader support memmap?

Reply:
In the file ana.py with PR #1750, the following line was added
```py
def read(filename, debug=False, **kwargs):```
So I thought I have to add a test for that.

Reply:
What happens to those kwargs in the read function?

Reply:
So when `memmap` is passed as a Keyword Arguments in the read function, this test will make sure that everything will stay fine using `assert`.

Reply:
Where do they end up?

Reply:
Sorry, I don't understand what you mean?

Reply:
In the docs here, https://docs.sunpy.org/en/stable/code_ref/io.html?highlight=io#id3
I don't find anything related to memmap, so do you think `ana.py` doesn't need this test?

Reply:
> In the docs here, https://docs.sunpy.org/en/stable/code_ref/io.html?highlight=io#id3 I don't find anything related to memmap, so do you think `ana.py` doesn't need this test?

cc @Cadair

Reply:
ANA most likely does not support memmap (you would have to check), but it's good to see that it doesn't error with the keyword set. 

You will need to replicate this for all the other readers and file types as well.

Reply:
In the PR [#1750](https://github.com/sunpy/sunpy/pull/1750) read function of 4 files was changed, they are:
- ana.py
- file_tools.py
- fits.py
- jp2.py

So, I will be writing tests for these four read functions.

Reply:
The documentation for [`sunpy.io.read_file`](https://docs.sunpy.org/en/latest/generated/api/sunpy.io.read_file.html#sunpy.io.read_file) says that `memmap` "is currently only supported by the FITS reader", i.e., `sunpy.io._fits.read` for which you have added a test which should work very nicely! (Although that test is currently erroring, maybe double check the type of object that `sunpy.io._fits.read` returns to see why it says [`not enough values to unpack (expected 2, got 1)`](https://github.com/sunpy/sunpy/runs/6231810970?check_suite_focus=true#step:8:5137).)

The rest of the changes to the read functions in that PR were to allow `memmap=` to be passed to the [`sunpy.io.read_file`](https://docs.sunpy.org/en/latest/generated/api/sunpy.io.read_file.html#sunpy.io.read_file) function and not have an error raised by a `sunpy.io.*.read` function if `memmap=` was passed but wasn't supported in it. 

So I guess the rest of the tests should just be to see that it doesn't raise an error if `memmap` is passed but unsupported. Although maybe sunpy should be presenting the user with a warning instead?

Reply:
> Although maybe sunpy should be presenting the user with a warning instead?

How do we know that the file supports memmap?

Reply:
You should be able to check each reader and see if they support memory mapping. As Conor has said, I would assume only the FITS reader supports it.